analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxKindFacade.cs

```diff
@@ -54,7 +54,7 @@ internal sealed class CSharpSyntaxKindFacade : ISyntaxKindFacade<SyntaxKind>
     public SyntaxKind ReturnStatement => SyntaxKind.ReturnStatement;
     public SyntaxKind SimpleAssignment => SyntaxKind.SimpleAssignmentExpression;
     public SyntaxKind SimpleMemberAccessExpression => SyntaxKind.SimpleMemberAccessExpression;
-    public SyntaxKind StringLiteralExpression => SyntaxKind.StringLiteralExpression;
+    public SyntaxKind[] StringLiteralExpressions => new[] { SyntaxKind.StringLiteralExpression, SyntaxKindEx.Utf8StringLiteralExpression };
```

Comment:
I am not sure this is a 100% correct, as UTF8 Strings are strings and at the same time they are not. I personally would not add the `Utf8StringLiteralExpression` here but rather handle them separately.

Reply:
From some tests I did - from an AST point of view, they seem to have similar behavior (almost the same).
Take a look at this [example in SharpLab](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLAChcoBYACAJTAHcBlYAJwEsA7AcwBk7gBTGsAGwgAoUGAAxEArgwDWDAPbkGABTA0YRIaOlzFy4AEpcAb1xETakUToAHDEQC8RAEROMATjQA6DADYAHO+GeTg4A3MamAG7KFpYAzHaOADoAHjEYyTEu6WjJaACi6WkpXuk+OfkpwmUFDmI+oTgAvkA=).

In the semantic universe, they are not the same - but I did not see any suspicious case that can get us bitten in the future so far.

Reply:
They might have similar behaviour and yet they are completely different things. And when I use `StringLiteralExpression` from the syntax kind facade I expect really to only handle string literals. There are rules and cases when we only want to handle one of them and not both. One of those examples could be the rule which says not to call `ToString` on a string literal as it is redundant. While it is redundant for the `StringLiteral` it is not for the `UTF-8 string literal`. And basically if one would want to implement that rule using the `SyntaxKindFacade` it would not be possible. I either suggest to either revert the Facade to its previous state and handle the SyntaxKind separately or Add a new property to the facade which clearly states that it contains both "StringLiterals". WDYT?

Reply:
I don't agree with Csaba here. We already face the same problem with interpolated strings, which can be of type string or FormattableString depending on the target type:
```cs
FormattableString fs = $""; 
string st = $""; 
```
We don't distinguish here also.

If you see a string literal, you can not be sure it is assigned to a `string` either:
```cs
object o = "";
IComparable c = "";
T t = "";

public class T
{
    public static implicit operator T(string v) => new T();
}
```
So you always have to check the target type and the only difference with `u8` is that the targets are now
for string literals:

* string
* object
* IComparable
* IEnumerable
* Any type with an implicit conversion from string
 
and for ut-8 literals:

* `ReadonlySpan<byte>`

Reply:
> One of those examples could be the rule which says not to call ToString on a string literal as it is redundant.

The rule would be to not call `ToString` on a string **type** not on a string **literal** (and besides that, there is a ToString overload on string which is sensible to call).

Reply:
@mary-georgiou-sonarsource You should have a look at where StringLiteralExpressions is used and add tests for ut8 literals in such rules. We can decide from there if @csaba-sagi-sonarsource concerns are valid today already.

Reply:
It seems that two Hotspots register for `StringLiteralExpressions`. 
I added UTs for utf-8 string literals to both.

Reply:
In the CSharp rules additionally, **S4784** `UsingRegularExpressions`  is using the "GetStringValue" - to check strings given as input to Regex Object creation or invocation.
I see no issue there regarding the facade update.

Reply:
Fine by me.

Reply:
> It seems that two Hotspots register for StringLiteralExpressions coming from the Facade.
I added UTs for utf-8 string literals to both.

I only see PublicWritableDirectories. What is the other one or do I miss something?

Reply:
Never mind. Its HardcodedIpAddress.